lib/active_merchant/billing/gateways/priority.rb

```diff
@@ -0,0 +1,347 @@
+        }
+      end
+
+      def request_verify_headers(jwt)
+        {
+          'Authorization' => "Bearer #{jwt}"
+        }
+      end
+
+      def purchase(amount, credit_card, options = {})
+        params = {}
+        params['amount'] = localized_amount(amount.to_f, options[:currency])
+        params['authOnly'] = false
+
+        add_credit_card(params, credit_card, 'purchase', options)
+        add_type_merchant_purchase(params, @options[:merchant_id], true, options)
+        commit('purchase', params: params, jwt: options)
+      end
+
+      def authorize(amount, credit_card, options = {})
+        params = {}
+        params['amount'] = localized_amount(amount.to_f, options[:currency])
+        params['authOnly'] = true
+
+        add_credit_card(params, credit_card, 'purchase', options)
+        add_type_merchant_purchase(params, @options[:merchant_id], false, options)
+        commit('purchase', params: params, jwt: options)
+      end
+
+      def refund(amount, credit_card, options)
+        params = {}
+        # refund amounts must be negative
+        params['amount'] = ('-' + localized_amount(amount.to_f, options[:currency])).to_f
+
+        add_bank(params, options[:auth_code])
+        add_credit_card(params, credit_card, 'refund', options) unless options[:auth_code]
+        add_type_merchant_refund(params, options)
+        commit('refund', params: params, jwt: options)
+      end
+
+      def capture(amount, authorization, options = {})
+        params = {}
+        params['amount'] = localized_amount(amount.to_f, options[:currency])
+        params['authCode'] = options[:authCode]
+        params['merchantId'] = @options[:merchant_id]
+        params['paymentToken'] = authorization
+        params['shouldGetCreditCardLevel'] = true
+        params['source'] = options['source']
+        params['tenderType'] = options[:tender_type]
+
+        commit('capture', params: params, jwt: options)
+      end
+
+      def void(iid, options)
+        commit('void', iid: iid, jwt: options)
+      end
+
+      def verify(credit_card, jwt)
+        commit('verify', card_number: credit_card, jwt: jwt)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def get_payment_status(batch_id, options)
+        commit('get_payment_status', params: batch_id, jwt: options)
+      end
+
+      def close_batch(batch_id, options)
+        commit('close_batch', params: batch_id, jwt: options)
+      end
+
+      def create_jwt(options)
+        commit('create_jwt', params: @options[:merchant_id], jwt: options)
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]').
+          gsub(%r((number\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r((cvv\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]')
+      end
+
+      def add_bank(params, auth_code)
+        params['authCode'] = auth_code
+        params['authMessage'] = ''
```

Comment:
Let's remove values we are defaulting to empty string or nil, so we can reduce some of the clutter in here. Unless it's explicitly stated in our SOW, I would rather have this boiled down to what is asked for this implementation.